pkg/systemd/services/service-details.jsx

```diff
@@ -308,6 +324,21 @@ export class ServiceDetails extends React.Component {
         }
     }
 
+    pinUnit(method) {
+        let pinnedUnits = JSON.parse(localStorage.getItem('pinnedUnits')) || [];
+
+        this.setState({ waitsAction: true });
+        if (method === "Pin") {
+            pinnedUnits.push(this.props.unit.path);
+        } else if (method === "Unpin") {
```

Comment:
I think passing "method" argument to the `pinUnit` is unnecessary. It can be a boolean, or alternatively, since this method is only called when the user wants to negate the "pinned" status of a service, this function doesn't need any argument at all, you can just look at the "pinned" status of the service and negate it.

Reply:
good point, fixed